File: tools/mixdown_kits.py


# TEST convert Drumgizmo drum kits to Edrumulus usage by mixdown all the audio channels
# using Ardour remote controlled by Open Sound Control (OSC)

# NOTE you need to activate OSC in Ardour in Edit->Preferences->Controller

from pythonosc import udp_client

ardour = udp_client.SimpleUDPClient("127.0.0.1", 3819) # Ardour default port number



#ardour.send_message("/strip/fader", [2, 0.5])


# not yet working: import wav file
#ardour.send_message("/Common/addExistingAudioFiles", "~/edrumulus/tools/DRSKit/Snare_circle_whisker/samples/4-Snare_circle_whisker.wav")

File: tools/mixdown_kits.lua

-- TEST convert Drumgizmo drum kits to Edrumulus usage by mixdown all the audio channels
-- using Ardour with lua scripting
-- see also: https://github.com/Ardour/ardour/blob/master/gtk2_ardour/luainstance.cc

ardour { ["type"] = "Snippet", name = "Mixdown Drumgizmo kits for Edrumulus" }

function factory ( params ) return function()

	local files = C.StringVector();

	files:push_back ( "/home/corrados/edrumulus/tools/DRSKit/Tom1_whisker/samples/11-Tom1_whisker.wav" )

	Editor:do_import (
		files,
		Editing.ImportDistinctChannels,
		Editing.ImportAsTrack,
		ARDOUR.SrcQuality.SrcBest,
		ARDOUR.MidiTrackNameSource.SMFTrackName,
		ARDOUR.MidiTempoMapDisposition.SMFTempoIgnore,
		-1,
		ARDOUR.PluginInfo() )


-- TODO how to export a stereo audio file...
	Editor:export_audio()


end end

